src/transformers/models/llama/modeling_llama.py

```diff
@@ -494,6 +496,11 @@ def forward(
             key_states = key_states.to(target_dtype)
             value_states = value_states.to(target_dtype)
 
+        batch_size = query_states.size(0)
+        query_states = query_states.reshape(-1, query_states.size(-2), query_states.size(-1))
+        key_states = key_states.reshape(-1, key_states.size(-2), key_states.size(-1))
+        value_states = value_states.reshape(-1, value_states.size(-2), value_states.size(-1))
```

Comment:
```suggestion
```
I think we should do this in the `_flash_attention_forward` wrapper this we have 0 modeling changes, and all models will benefit easily from this!